fpdf/text_region.py

```diff
@@ -0,0 +1,275 @@
+from .errors import FPDFException
+from .enums import Align, XPos, YPos
+from .line_break import MultiLineBreak
+
+# Since Python doesn't have "friend classes"...
+# pylint: disable=protected-access
+
+
+class TextRegionMixin:
+    """Mix-in to be added FPDF() in order to support text regions."""
+
+    def __init__(self, *args, **kwargs):
+        self.clear_text_region()
+        super().__init__(*args, **kwargs)
+
+    def register_text_region(self, region):
+        self.__current_text_region = region
```

Comment:
Why using a double-underscore property there?

There handled in a special way by Python: https://stackoverflow.com/a/1301369/636849

Reply:
Maybe it's overkill, but it will never collide with any other attribute! :wink:
As far as I can tell, this is exactly the intended use case for this language feature.